Shared/Store/BaseDataStore.swift

```diff
@@ -157,26 +157,19 @@ class BaseDataStore {
                 })
                 .disposed(by: self.disposeBag)
 
-        dispatcher.register
-                .filterByType(class: LifecycleAction.self)
-                .subscribe(onNext: { action in
-                    switch action {
-                    case .background:
-                        self.autoLockSupport.storeNextAutolockTime()
-                    case .foreground:
-                        self.initializeLoginsStorage()
-                        self.handleLock()
-                    case .upgrade(let previous, _):
-                        if previous <= 2 {
-                            self.handleLock()
-                        }
-                    case .shutdown:
-                        self.shutdown()
-                    default:
-                        break
-                    }
```

Comment:
I understand replacing the `filterByType` with the `lifeCycleStore.lifecycleEvents`.

I don't understand how/why you can get rid of _all_ of the `switch` statement. Especially the `upgrade` actions.

Reply:
A lot of them got moved into the `setupAutoLock` method, and then it seemed like overkill to have a switch statement for two cases so I split those out into their own subscriptions, although I think that's more of a personal preference than a design requirement.

re: removing the `upgrade` code, I'm OK with cleaning up that dead code as >99% of our active users have upgraded to a higher version already, and the workaround for missing the upgrade code is that they will have to log in again.

Reply:
:+1: